src/algorithm/navigation_algorithm.py

```diff
@@ -27,7 +41,13 @@ def generate_random_grid(self, rows, cols):
                 else:
                     self.grid[i, j] = 1
 
-    def a_star_search(self):
+    def a_star_search(self) -> list:
```

Comment:
```suggestion
    def a_star_search(self) -> list[tuple]:
```

w zależności od użytej wersji Pythona (>= 3.10.14) powinno działać bez zarzutu